### ISSUE TITLE
Add benchmark command to measure model serving performance

### ISSUE THREAD

**Issue by peshwar9** [sudhanvalabs/mlship#1]
## Feature Request

  Add a `mlship benchmark` command to measure latency and throughput of served models.

  ## Motivation

  Users need to understand the performance characteristics of their models when served via mlship. Currently, they have to write custom scripts to measure this. A built-in benchmark command would:

  - Help users evaluate if mlship meets their performance requirements
  - Provide consistent benchmarking methodology across different models
  - Make it easy to compare performance across frameworks (sklearn vs PyTorch vs TensorFlow)
  - Generate shareable performance reports

  ## Proposed Solution

  Add a new CLI command:

  ```bash
  mlship benchmark model.pkl [options]

  Options:
  - --requests N - Number of requests to run (default: 100)
  - --warmup N - Number of warmup requests (default: 5)
  - --port PORT - Port to use (default: 8000)
  - --payload '{"features": [...]}' - Custom test payload (optional)

  Workflow:
  1. Start mlship server in background
  2. Wait for server to be ready
  3. Measure cold start latency
  4. Run warmup requests
  5. Run benchmark requests and collect latencies
  6. Display results table
  7. Stop server and cleanup

  Output:

  Benchmarking sklearn_model.pkl...
  ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
  Cold Start:    1.234s
  Warmup:        5 requests
  Benchmark:     100 requests
  ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  Performance Metrics:
    Average:       45.23ms
    Min:           32.10ms
    P50 (Median):  44.50ms
    P95:           67.80ms
    P99:           89.20ms
    Max:           102.30ms

  Throughput:    ~22 requests/sec
  ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  Implementation Notes

  - Should work with all supported frameworks (sklearn, PyTorch, TensorFlow, HuggingFace)
  - Auto-generate appropriate test payload if not provided
  - Handle server startup/shutdown gracefully
  - Consider adding JSON output option for CI/CD integration: --output json

  Alternatives Considered

  - External benchmark script (user burden, inconsistent methodology)
  - Only document how to benchmark (doesn't provide standardized approach)

  Additional Context

  This would be great for:
  - Performance regression testing
  - Comparing different model formats
  - Generating performance data for documentation
  - Helping users make informed deployment decisions

  Example use case:
  # Benchmark different model types
  mlship benchmark sklearn_model.pkl --requests 1000
  mlship benchmark pytorch_model.pt --requests 1000
  mlship benchmark tensorflow_model.h5 --requests 1000

**Comment by peshwar9** [comment]
I'm working on implementing this feature now. Should have a PR ready in the coming days. If anyone wants to collaborate or has suggestions on the implementation, feel free to comment.
